### ISSUE TITLE
isinstance and issubclass should not create intersections if one of the types is final

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5016]
This is related to [this bug report](https://github.com/python/mypy/issues/15148) in the mypy type tracker.

```python
class A: ...

@final
class D: ...

def func4(val: D):
    if isinstance(val, A):
        reveal_type(val, expected_text="Never") # Currently produces <subtype of A and D>
```

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.306, which I just published. It will also be included in a future release of pylance.
